docs/developer/index.rst

```diff
@@ -1136,11 +1166,11 @@ We can view the model plot object::
     >>> print(plot)
     xlo    = [2,3,5,7,8]
     xhi    = [3,5,6,8,9]
-    y      = [ 8.5,20. ,11.5,13.5,14.5]
+    y      = [ 6.,12., 6., 6., 6.]
     xlabel = x
     ylabel = y
     title  = Model
-    histo_prefs = {'yerrorbars': False, 'ecolor': None, ... , 'linecolor': None}
+    histo_prefs = {'xerrorbars': False, 'yerrorbars': False, ..., 'linecolor': None}
```

Comment:
Hmm. dictionary ordering has calmed down from earlier python versions, so I think it's now safe to assume a given order.

Reply:
Yes. Always in insertion order now.

Reply:
Of course, the way we create any particular dict could change for a variety of reasons, so we may see changes over time (and would have to think about at that time).

Does doctest treat

    {'a': False, 'b': True}
    {'b': True, 'a': False}

the same? For the line highlighted here it may be hard to check since we elide some of the keys, so it would be hard to check.

Reply:
Doctest performs a string comparison of the output. So, two identical dicts can fail if printed in different order. Luckily, now the order of dict items is deterministic in Python, as long as they are generated in a deterministic fashion (which they are for this case). Again, it's a trade-off: Do we want to made doctest happy? Is it worth the extra work or should we just ignore this test or entire output line? Or should we take the fact that the output line is super-long (> 300 characters I think) as a sign that the output is not great to begin with an should be changed?

It's something I don't have a good feeling for yet, but that we'll get experience with as we develop this further. 

In some instances, it might be useful to write code that does not depend on the order e.g. `print(histoprefs['xlog'])` as an example what's in there instead of `print(histoprefs)` or `print(sorted(histroprefs))`; in others we want to show the users how the entire dict looks when printed on screen and then it might be better to just not check that output.

First priority of a docstring should be that it's useful for the reader (which means it's also correct and executes without error). Using it as unit test is a lower priority to me, but does not hurt if we can get it to work with minimal extra effort and without reducing the usefulness of the docs for the reader.